index.js

```diff
@@ -146,6 +147,13 @@ function session(options){
     if (cookieOptions.secure === 'auto') {
       req.session.cookie.secure = issecure(req, trustProxy);
     }
+
+    if (cookieOptions.autoSubDomain) {
+      var parsedDomain = parseDomain(req.headers.host);
+      if (parsedDomain && parsedDomain.domain) {
+        req.session.cookie.domain = '.' + parsedDomain.domain + '.' + parsedDomain.tld;
```

Comment:
I'm curious here: why is the option called `autoSubDomain`, but yet this just leaves the `parsedDomain.subdomain` out of the picture? Perhaps I'm just not understanding what you're trying to do here. If you add the documentation for your feature to the README, I can read that and see if I understand :)